From f710f9f1a59e5786953c82fa1d258dda419821d5 Mon Sep 17 00:00:00 2001
From: MD Danish Anwar <danishanwar@ti.com>
Date: Fri, 23 Jun 2023 18:03:31 +0530
Subject: [PATCH 078/143] irqchip: irq-ti-sci-inta: Add null check for
 parent_irq_data

Add null check for parent_irq_data in ti_sci_inta_set_affinity() API.
Without null check previously, parent_irq_data was getting de-referenced
resulting in null pointer de-reference Kernel crash.

Signed-off-by: MD Danish Anwar <danishanwar@ti.com>
---
 drivers/irqchip/irq-ti-sci-inta.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/drivers/irqchip/irq-ti-sci-inta.c b/drivers/irqchip/irq-ti-sci-inta.c
index f965dc59d81c..f2225e27e212 100644
--- a/drivers/irqchip/irq-ti-sci-inta.c
+++ b/drivers/irqchip/irq-ti-sci-inta.c
@@ -526,6 +526,10 @@ static int ti_sci_inta_set_affinity(struct irq_data *d,
 	event_desc = irq_data_get_irq_chip_data(d);
 	if (event_desc) {
 		vint_desc = to_vint_desc(event_desc, event_desc->vint_bit);
+		parent_irq_data = irq_get_irq_data(vint_desc->parent_virq);
+
+		if (!parent_irq_data || !parent_irq_data->chip->irq_set_affinity)
+			return -EINVAL;
 
 		/*
 		 * Cannot set affinity if there is more than one event
@@ -537,9 +541,8 @@ static int ti_sci_inta_set_affinity(struct irq_data *d,
 		vint_desc->affinity_managed = true;
 
 		irq_data_update_effective_affinity(d, mask_val);
-		parent_irq_data = irq_get_irq_data(vint_desc->parent_virq);
-		if (parent_irq_data->chip->irq_set_affinity)
-			return parent_irq_data->chip->irq_set_affinity(parent_irq_data, mask_val, force);
+
+		return parent_irq_data->chip->irq_set_affinity(parent_irq_data, mask_val, force);
 	}
 
 	return -EINVAL;
-- 
2.41.0

